rodeo/src/rodeo/06-either-and-try.scala

```diff
@@ -0,0 +1,186 @@
+      else if (a == 0)
+        Failure(
+          new IllegalArgumentException("This is not a quadratic equation")
+        )
+      else
+        Failure(
+          new IllegalArgumentException(
+            "This equations does not have a solution in the real numbers"
+          )
+        )
+    }
+
+    assertTrue(quadraticEquationSolver(0, 7, 13) == ???)
+    assertTrue(quadraticEquationSolver(???, ???, ???) == Success(Math.sqrt(2)))
+  }
+
+  Exercise("Party") {
+    case class Friend(name: String, isClose: Boolean)
+
+    val ana = Friend("Ana", true)
+
+    case class Party(title: String, isBig: Boolean)
+
+    val christmasParty = Party("Christmas", true)
+
+    case class Invitation(friend: Friend, party: Party)
+
+    def invitation(friend: Friend, party: Party): Try[Invitation] = {
+      if (friend.isClose) Success(Invitation(friend, party))
+      else if (party.isBig) Success(Invitation(friend, party))
+      else
+        Failure(
+          new IllegalStateException(
+            "Friend ${friend.name} is not a close friend and this is a small party}"
+          )
+        )
+    }
+
+    assertTrue(invitation(ana, christmasParty) == ???)
+
+  }
+
+  // We can define functional methods in Try[A] like map, filter and flatmap. This methods behave in almost the same way as they do in the case of Options.
+
+  Exercise("map method on Try") {
+    val names = List(
+      Success("David"),
+      Success("Vic"),
+      Success("Fabian"),
+      Failure(new IllegalArgumentException("Andrew"))
+    )
+    val namesOffice = names.map {
+      case Failure(name) => "Lacayo rebelde"
+      case Success(name) => Success(name + "_" + "DealEngine")
+    }
+
+    assertTrue(namesOffice(1) == ???)
+  }
+
+  // We will now introduce the `Either[A,B]` type.
+  // An instance of the `Either[A,B]`type is either an instance of type A or an instance of type B. In mathematical terms `Either[A,B]` is the disjoint union of A and B.
+  // `Either` can be used in cases where a computation could return two different types depending on a certain condition imposed in the input.
+  // The `Either[A,B]` type is more general than the `Try[A]` type. Try[A] = Either[Failure[A],Success[A]]`.
+  // One difference in the case of exception handling between Try and Either is that by using Either we are not forced to represent failure by a wrapped instance of the Throwable class.
+  // We can understand `Either[A,B]` intuitively as a choice between left and right. For exception handling, left represents failure while right represents success.
+
+  Exercise("Tacos") {
+    def numberOfTacos(
+        weight: Double,
+        exercised: Boolean
+    ): Either[String, Double] =
+      if (weight < 0) Left("This is not a valid weight")
+      else Right(if (exercised) 2 * (weight / 20) else weight / 20)
+
+    assertTrue(numberOfTacos(80, false) == ???)
+  }
+
+  Exercise("SquareRoot") {
+    def sqrt(x: Double): Either[String, Double] =
```

Comment:
I like this exercise, perhaps we should use it as well but with `Try[Double]`? instead of the discriminant one?